front/src/common/redux/auth/auth.api.js

```diff
@@ -8,4 +8,35 @@ async function loginApi(request) {
     return axios.post(url, data, headers)
         .then(response => response)
 };
-export { loginApi };
\ No newline at end of file
+
+async function changePasswordApi(request) {
+    const token = localStorage.getItem('token');
+    const teacherURL = process.env.REACT_APP_CHANGE_TEACHER_PASSWORD;
+    const studentURL = process.env.REACT_APP_CHANGE_STUDENT_PASSWORD;
+    const { oldPassword, newPassword } = request.user
+    const headers = {
+        headers: {
+            "Accept": "application/json",
+            'Content-Type': 'application/json',
+            'Authorization': `Bearer ${token}`,
+        }
+    };
+
+    if (request.user.hasOwnProperty('teacherId') && request.user.hasOwnProperty('admin')) {
```

Comment:
This logic should be inside the saga function. API functions should just simply make HTTP request and return the response, nothing more

Reply:
done